jax/interpreters/xla.py

```diff
@@ -606,15 +627,17 @@ def _xla_callable(fun: lu.WrappedFun, device, backend, name, donated_invars, *ar
   device = _xla_callable_device(nreps, backend, device, arg_devices)
   backend = device.platform if device else backend
   if config.omnistaging_enabled:
-    result_handlers = tuple(aval_to_result_handler(device, a) for a in out_avals)
+    nouts = [aval._num_buffers for aval in out_avals]
+    result_handlers = map(partial(aval_to_result_handler, device), out_avals)
   else:
-    result_handlers = tuple(map(partial(_pval_to_result_handler, device), pvals))
+    nouts = [1 if pval is None else pval._num_buffers for pval, _ in pvals]
```

Comment:
```suggestion
    nouts = [pval.get_aval()._num_buffers for pval in pvals]
```

This code was buggy before; it assumed that if an output was known at trace time (meaning `pv is None` where `pv, const = pval`) then it must correspond to a single buffer. But actually we need to check the number of buffers regardless of whether the output is known or not.

(In partial_eval.py, we usually follow the convention of referring to the first element of a `pval` as `pv`, so if we wanted to unpack `pvals` we should do it like `for pv, const in pvals`, whereas writing `for pval, _ in pvals` maeks it seem like the first element of a pval is itself a pval.)